### ISSUE TITLE
Boundary layer mesh generation, prisms/wedges

### ISSUE THREAD

**Issue by dyollb** [CGAL/cgal#103]
It would be nice if CGAL would include a robust function to generate boundary layer meshes from triangle surface meshes. The preferred cell type to be generated are prisms/wedges. The user could specify the thickness, or a scalar field describing the thickness of the layer. The function should be able to deal with non-manifold surface meshes, and avoid self-intersections.

**Comment by sloriot** [comment]
Is this what the offset meshing plugin is doing?

**Comment by MaelRL** [comment]
No, boundary layer meshes are obtained by extruding (multiple times) the mesh of the surface, with vertex movement and potentially collapse/additions to avoid self-intersections.

**Comment by sloriot** [comment]
I understand, meshing different iso-level of the distance function to a triangle mesh. But this should be easily implementable with the code of the plugin.
